### PR TITLE
Download code

### PR DESCRIPTION
**Pull request by oliferwolox**
## Summary

- Add download code button.
- Add library JSZip https://github.com/Stuk/jszip
- Add library FileSaver https://github.com/eligrey/FileSaver.js

## Screenshots

download-code

## Trello Card

https://trello.com/c/9WblHLM5/46-descarga-de-componentes

### CODE REVIEW

File: src/app/screens/Dashboard/screens/Detail/index.tsx [hunk 1]
```diff
@@ -11,8 +13,20 @@ function Detail() {
   const { category, component } = useParams();
   const { data, loading } = useQuery(getComponentFiles(category as string, component as string));
   const componentCode = getComponentCode(data?.repository);
+  
+  const downloadZip = useCallback(() => {
+    const { html, scss, css } = componentCode;
```
Comment (mcallegari10) [inline src/app/screens/Dashboard/screens/Detail/index.tsx:18]: I think you should loop through the keys of the `componentCode`, that way you can get all files and that should support React code as well (maybe a few tweaks will be needed 😝)

File: src/app/screens/Dashboard/screens/Detail/index.tsx [hunk 2]
```diff
@@ -11,8 +13,20 @@ function Detail() {
...
+    var zip = new JSZip();
```
Comment (mcallegari10) [inline src/app/screens/Dashboard/screens/Detail/index.tsx:19]:
```suggestion
    const zip = new JSZip();
```

File: src/app/screens/Dashboard/screens/Detail/index.tsx (hunk 1 above)
Comment (oliferwolox) [reply src/app/screens/Dashboard/screens/Detail/index.tsx:18]: How do i manage the extension in the file?

Comment (mcallegari10) [reply src/app/screens/Dashboard/screens/Detail/index.tsx:18]: Touchè. Maybe you can use the key name? Ok, somehow we need to add the name to the files so.. that's something to be added later on. For now it's ok I think 😄

**Pull request merged**